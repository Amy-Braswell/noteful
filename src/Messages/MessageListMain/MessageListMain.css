
.MessageListMain ul {
    list-style-type: none;
    margin: 0;
    padding: 10px;
  }
  
  .MessageListMain__button-container {
    max-width: 380px;
    margin: 0 10px;
  }
  
  .MessageListMain__add-message-button {
    margin: 0 auto 20px;
  }
  .MessageListMain__add-message-button:hover {
    background-color: #29862e;
    color: white;
  }
  