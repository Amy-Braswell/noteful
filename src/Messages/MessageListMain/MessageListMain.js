import React from 'react'
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import { Link } from 'react-router-dom'
import ApiContext from '../../ApiContext'
import CircleButton from '../../CircleButton/CircleButton'
import Message from '../Message/Message'
import { getMessagesForMember } from '../../messages-helpers'
import './MessageListMain.css'

export default class PostListMain extends React.Component {
  static defaultProps = {
    match: {
      params: {}
    }
  }
  static contextType = ApiContext

  
  render() {
    const { memberid } = this.props.match.params
    const { messages=[] } = this.context
    const messagesForMember = getMessagesForMember(messages, memberid)
    return (
      <main className='PostListMain'>
        <ul>
          {messagesForMember.map(message =>
            <li key={message.id}>
              <Message
                id={message.id}
                content={message.content}
                name={message.name}
                modified={message.modified}
              />
            </li>
          )}
        </ul>
        <div className='MessageListMain__button-container'>
          <CircleButton 
            tag={Link}
            to='/add-message'
            type='button'
            className='MessageListMain__add-message-button'
          >
            <FontAwesomeIcon icon={['fa', 'plus']} />
            <br />
            Message
          </CircleButton>
        </div>
      </main>
    )
  }
}






// export default class MessageListMain extends React.Component {
//   static defaultProps = {
//     match: {
//       params: {}
//     }
//   }
//   static contextType = ApiContext

  
//   render() {
//     const { memberid } = this.props.match.params
//     const { messages=[] } = this.context
//     const messagesForMember = getMessagesForMember(messages, memberid)
//     return (
//       <main className='MessageListMain'>
//         <ul>
//           {messagesForMember.map(message =>
//             <li key={message.id}>
//               <Message
//                 id={message.id}
//                 content={message.content}
//                 name={message.name}
//                 modified={message.modified}
//               />
//             </li>
//           )}
//         </ul>
//         <div className='MessageListMain__button-container'>
//           <CircleButton 
//             tag={Link}
//             to='/add-message'
//             type='button'
//             className='MessageListMain__add-message-button'
//           >
//             <FontAwesomeIcon icon={['fa', 'plus']} />
//             <br />
//             Message
//           </CircleButton>
//         </div>
//       </main>
//     )
//   }
// }
