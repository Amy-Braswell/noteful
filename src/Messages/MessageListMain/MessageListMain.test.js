import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import MessageListMain from './MessageListMain'

describe(`MessageListMain component`, () => {
  it('renders a .MessageListMain by default', () => {
    const wrapper = shallow(<MessageListMain />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  // enzyme doesn't yet support React.createContext
  it.skip('renders a Message in ul for each messages in array', () => {
    const props = {
      match: {
        params: {
          memberid: 'THIS_MEMBER_ID'
        }
      }
    }
    const context = {
      messages: [
        {
          "id": "cbc787a0-ffaf-11e8-8eb2-f2801f1b9fd1",
          "name": "This is a test message about Dogs",
          "modified": "2019-01-03T00:00:00.000Z",
          "memberid": "b0715efe-ffaf-11e8-8eb2-f2801f1b9fd1",
          "content": "Corporis accusamus placeat.\n \rUnde."
        },
        {
          "id": "d26e0034-ffaf-11e8-8eb2-f2801f1b9fd1",
          "name": "This is a test message about Cats",
          "modified": "2018-08-15T23:00:00.000Z",
          "memberid": "THIS_MEMBER_ID",
          "content": "Eos\n \rlaudantium."
        },
        {
          "id": "d26e01a6-ffaf-11e8-8eb2-f2801f1b9fd1",
          "name": "This is a test message about Pigs",
          "modified": "2018-03-01T00:00:00.000Z",
          "memberid": "THIS_MEMBER_ID",
          "content": "Occaecati dignissimos\nvoluptatum nihil."
        },
        {
          "id": "d26e0570-ffaf-11e8-8eb2-f2801f1b9fd1",
          "name": "This is a test message about Birds",
          "modified": "2019-01-04T00:00:00.000Z",
          "memberid": "b0715efe-ffaf-11e8-8eb2-f2801f1b9fd1",
          "content": "Eum culpa odit."
        },
      ]
    }
    const ul = shallow(<MessageListMain {...props} />, context)
      .find('ul')
    expect(toJson(ul)).toMatchSnapshot()
  })
})