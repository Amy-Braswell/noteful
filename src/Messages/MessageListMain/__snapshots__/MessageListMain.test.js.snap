// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MessageListMain component renders a .MessageListMain by default 1`] = `
<main
  className="PostListMain"
>
  <ul />
  <div
    className="MessageListMain__button-container"
  >
    <NavCircleButton
      className="MessageListMain__add-message-button"
      tag={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "displayName": "Link",
          "propTypes": Object {
            "innerRef": [Function],
            "onClick": [Function],
            "replace": [Function],
            "target": [Function],
            "to": [Function],
          },
          "render": [Function],
        }
      }
      to="/add-message"
      type="button"
    >
      <FontAwesomeIcon
        border={false}
        className=""
        fixedWidth={false}
        flip={null}
        icon={
          Array [
            "fa",
            "plus",
          ]
        }
        inverse={false}
        listItem={false}
        mask={null}
        pull={null}
        pulse={false}
        rotation={null}
        size={null}
        spin={false}
        swapOpacity={false}
        symbol={false}
        title=""
        transform={null}
      />
      <br />
      Message
    </NavCircleButton>
  </div>
</main>
`;
